Implemented Simple Unit Movement for first ProtoType

ChristianBasiga opened the pull request:
I edited Tile script here, which is part of map, these edits were for
testing and will be mostly discarded. Leaving in commit for review
incase you want to keep any of the edits. In future will write in
separate temp script/ask people developing the map section to write in
what I need.

Commit:
Implemented Simple Unit Movement for first ProtoType

I edited Tile script here, which is part of map, these edits were for
testing and will be mostly discarded. Leaving in commit for review
incase you want to keep any of the edits. In future will write in
separate temp script/ask people developing the map section to write in
what I need.

kodomazer commented on Phalanx/Assets/Scripts/Entity/IOccupant.cs:
```diff
@@ -9,7 +9,7 @@ public interface IOccupant
         Map.Tile tile { get; set; }
         AttackInfo attack { get; }
-
+        bool selected { get; set; }
```
The occupant doesn't know if it's selected or not.

kodomazer commented on Phalanx/Assets/Scripts/Entity/Unit.cs:
```diff
@@ -7,43 +7,86 @@ namespace Olympus.Phalanx.Entity
+                Debug.Log("Moving");
+                float timeSinceLastMovement = Time.time - timeStartedMovement;
+                float distanceToMovePerFrame = timeSinceLastMovement / distanceToNextTile;
+                transform.position = Vector3.Lerp(tile.transform.position, destination.transform.position, distanceToMovePerFrame);
```
You're supposed to give the fraction of the linear interpolation that has elapsed.

For example: if the lerp is supposed to take one second then half a second into it you pass it 0.5.

kodomazer commented on Phalanx/Assets/Scripts/Entity/Unit.cs:
```diff
@@ -7,43 +7,86 @@ namespace Olympus.Phalanx.Entity
+        public void move()
+        {
+            //Why does is it giving me null reference? I break pointed to it and clearly is there and correct val
+            if (transform.position != destination.transform.position)
```
You probably need to make a check for if destination is a valid gameObject
Possibly just make destination a Vector3 in it of itself.

kodomazer commented on Phalanx/Assets/Scripts/Entity/Unit.cs:
```diff
@@ -7,43 +7,86 @@ namespace Olympus.Phalanx.Entity
+        public bool readyToMove
+        {
+            //This set is only for outsiders when rdy to move.
+            set
```
Put the info from this set into the set for the tile, that way the moment the tile is assigned then it knows to move to the new tile.

kodomazer reviewed (changes requested):
Some of the implementation in Unit needs to be fixed or shifted around.

Also, try to avoid committing changes to portions of code that other people are likely working on.  If you need extra functions in some other class add in the call as a comment and tell them about it, that way we can have cleaner pulls (hopefully)

Commit:
Made changes/fixes to Unit and got rid of changes to Tile

Also commiting change to my git ignore for test temp files. That way I
can test if things work without effecting actual files.

Commit:
Commiting git ignore list update

In hindsight better to just make folder and ignoring that, rather than
just ignoring specific file

kodomazer commented:
If you do the changes and then commit them then they will be added to the pull request.

Commit:
Applied changes to Unit and discarded my previous changes to Tile.

kodomazer reviewed (approved):
Looks good.  I'll shift things around once I get the Map code and start integrating things.
